### PR TITLE
[LLVMGPU] Add conversion to ldmatrix in distribution 

### PR DESCRIPTION
**Pull request by ThomasRaoux**
Emit ldmatrix when the layout is compatible.

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/LLVMGPU/Utils/LLVMGPULayoutAnalysisAndDistribution.cpp [hunk 1]
```diff
@@ -411,6 +411,94 @@ static SmallVector<Value> getDistributedIndices(
...
+  if (map.getNumResults() != 2) {
+    return false;
+  }
+  auto exprX = map.getResult(0).dyn_cast<AffineDimExpr>();
+  auto exprY = map.getResult(1).dyn_cast<AffineDimExpr>();
+  if (!exprX || !exprY) return false;
+  return exprX.getPosition() > exprY.getPosition();
+}
+
+/// Emit ldmatrix op code sequence.
```
Comment (harsh-nod) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/Utils/LLVMGPULayoutAnalysisAndDistribution.cpp]: Could you add a few more comments explaining this function?

Comment (ThomasRaoux) [reply compiler/src/iree/compiler/Codegen/LLVMGPU/Utils/LLVMGPULayoutAnalysisAndDistribution.cpp]: Done, hopefully this is easier to understand.

Comment (harsh-nod) [reply compiler/src/iree/compiler/Codegen/LLVMGPU/Utils/LLVMGPULayoutAnalysisAndDistribution.cpp]: Yes that helps. Thanks!

**Review by harsh-nod — APPROVED** [review]
Looks good! Thanks :)

**Pull request merged**